### PR TITLE
026automatic auditing

### PR DESCRIPTION
**Pull request by OlgaEvdokimova**
add automatic auditing

### CODE REVIEW

**Review by ShirshovAlex — DISMISSED** [review]

File: src/main/java/by/exadel/internship/auditing/Auditable.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.data.annotation.LastModifiedBy;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.Column;
+import javax.persistence.EntityListeners;
+import javax.persistence.MappedSuperclass;
+import java.time.LocalDate;
+
+@Getter(AccessLevel.PROTECTED)
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/auditing/Auditable.java]: Why protected?

File: src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+databaseChangeLog:
+  - changeSet:
+      id: addColumn-example
```
Comment (mortal-gpt) [inline src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml]: correct id name please

File: src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      author: liquibase-docs
```
Comment (mortal-gpt) [inline src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml]: correct author, please

**Comment by mortal-gpt** [comment]
@OlgaEvdokimova 
I don't think this file - OlgaEvdokimova_Resume.docx should be in the repository

File: src/main/java/by/exadel/internship/auditing/Auditable.java (hunk 1 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/auditing/Auditable.java]: To use getters or setters only with the inherited classes, for instance Internship extends Auditable<String>

File: src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml (hunk 2 above)
Comment (OlgaEvdokimova) [reply src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml]: done

File: src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml (hunk 3 above)
Comment (OlgaEvdokimova) [reply src/main/resources/db/changelog/v.0.3/db.changelog-add-columns-auditing-internship.yaml]: done

File: src/main/java/by/exadel/internship/auditing/Auditable.java (hunk 1 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/auditing/Auditable.java]: deleted and put annotation Data

File: src/main/java/by/exadel/internship/auditing/Auditable.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import javax.persistence.Column;
+import javax.persistence.EntityListeners;
+import javax.persistence.MappedSuperclass;
+import java.time.LocalDate;
+
+@Data
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public  class Auditable<U> {
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/auditing/Auditable.java]:
```suggestion
public abstract class Auditable<U> {
```

Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/auditing/Auditable.java]: done

**Review by ShirshovAlex — APPROVED** [review]

**Review by viktarkolbik — APPROVED** [review]

**Review by kzaretskaya — APPROVED** [review]

**Pull request merged**